### PR TITLE
feat: adpat new tailwindcss tokens

### LINKED ISSUE #80: Update tailwindcss tokens for Minimals
https://minimals.cc/components

- [x] [Color](https://minimals.cc/components/color) #79
- [x] [Typography](https://minimals.cc/components/typography) #79
- [x] [Shadows](https://minimals.cc/components/shadows) #79
- [x] [Grid/Spacing](https://minimals.cc/components/grid) #79
- [x] [Icons](https://minimals.cc/components/icons) (Migrate icons to [@iconify/react](https://docs.iconify.design/icon-components/react/))  
- [x] Fix broken tokens

### PR DESCRIPTION
**Pull request by Andrewnt219**
Resolves #80 .

You can read the changes if you want, but I recommend only checking the new UI from preview deployment

### CODE REVIEW

File: src/ui/Text/Text.tsx [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
-  as?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'p';
+  component?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'p' | 'span';
   variant?: StyledTextProps['variant'];
 };
 function Text({ className, ...props }: PropsWithChildren<Props>) {
   return (
     <StyledText
-      as={props.as ?? 'p'}
+      {...props}
+      as={props.component ?? 'span'}
```
Comment (vitokhangnguyen) [inline src/ui/Text/Text.tsx:13]: Default should be `<p>` tag, it's just more intutitive.

**Review by vitokhangnguyen — CHANGES_REQUESTED** [review]
This is a lot of work. Great job!

File: @types/twin.d.ts [hunk 2]
```diff
@@ -1,5 +1,7 @@
 /* eslint-disable no-undef */
 // twin.d.ts
+import { Icon } from '@iconify/react';
+import { ComponentProps, VFC } from 'react';
```
[the remark this answers is not in the document]
Comment (Andrewnt219) [reply @types/twin.d.ts:4]: Ah, this is from the 1h debug.

File: package.json [hunk 3]
```diff
@@ -24,19 +24,20 @@
     "firebase": "^8.7.1",
     "firebase-admin": "^9.11.0",
     "lodash": "^4.17.21",
+    "nanoid": "^3.1.30",
```
[the remark this answers is not in the document]
Comment (Andrewnt219) [reply package.json:27]: Doesn't seem like we use it anywhere.

[the remark this answers is not in the document]
Comment (vitokhangnguyen) [reply package.json:27]: Then we should remove it

File: src/modules/account/components/AccountMenu/AccountMenu.tsx [hunk 4]
```diff
@@ -48,14 +48,14 @@ function getMenuItems(t: Translate): MenuItem[] {
       data: {
         title: t('account:index.menu-items.personal-info.title'),
         description: t('account:index.menu-items.personal-info.description'),
-        icon: <FaInfoCircle tw="text-secondary" />,
+        icon: <Icon icon="mdi:information" tw="text-secondary" />,
```
[the remark this answers is not in the document]
Comment (Andrewnt219) [reply src/modules/account/components/AccountMenu/AccountMenu.tsx:51]: Eva, sure.

Why that syntax?

File: src/ui/Modal/Modal.tsx [hunk 5]
```diff
@@ -94,28 +100,22 @@ const modalDialogCss = (size: ModalSize) => {
 
   return css`
     max-width: calc(100% - 1rem);
-    top: calc(50% - ${theme('spacing.var-app-bar')} / 2);
```
[the remark this answers is not in the document]
Comment (Andrewnt219) [reply src/ui/Modal/Modal.tsx:97]: I think I fixed that already.

File: src/ui/Modal/Modal.tsx [hunk 6]
```diff
@@ -94,28 +100,22 @@ const modalDialogCss = (size: ModalSize) => {
...
-const ModalHeaderContent = styled.div`
-  ${tw`overflow-auto max-h-16`}
 `;
 
 const ModalBody = styled.div`
-  ${tw`p-xl`}
+  ${tw`px-lg pb-xl pt-lg`}
 `;
 const ModalBodyContent = styled.div`
-  max-height: calc(100vh - ${theme`spacing.var-app-bar`} - 8rem);
```
[the remark this answers is not in the document]
Comment (Andrewnt219) [reply src/ui/Modal/Modal.tsx:118]: I think I fixed that already.

File: src/modules/account/components/AccountMenu/AccountMenu.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (vitokhangnguyen) [reply src/modules/account/components/AccountMenu/AccountMenu.tsx:51]: Importing and passing it in seems better than passing string? 🤔  Unless when we type the string, the library provide intellisense?

File: src/ui/Modal/Modal.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (vitokhangnguyen) [reply src/ui/Modal/Modal.tsx:97]: I'll take your words for it

File: src/modules/account/components/AccountMenu/AccountMenu.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (Andrewnt219) [reply src/modules/account/components/AccountMenu/AccountMenu.tsx:51]: That's extra work though, it's also a string when we specify the import so I see no difference in risks.

File: src/modules/account/components/AccountInfoCard/AccountInfoCard.tsx [hunk 7]
```diff
@@ -11,11 +12,14 @@ type Props = {
 function AccountInfoCard({ className, ...props }: PropsWithChildren<Props>) {
   return (
     <StyledWrapper className={className} tw="">
-      <header tw="text-lg">
-        <span aria-hidden css={{ fontSize: '1.75em' }}>
+      <header>
+        <Text component="span" aria-hidden tw="text-[length:2.5em]">
```
Comment (vitokhangnguyen) [inline src/modules/account/components/AccountInfoCard/AccountInfoCard.tsx:16]: I'm thinking if it's not actually text, let's not use the `<Text>` componenet

**Review by vitokhangnguyen — CHANGES_REQUESTED** [review]
Styling suggestion: The shadow of the navbar should be more prominenet. Besides that, the UI quality seems impreoved by a lot with thje new sstyle system

image

image

**Comment by vitokhangnguyen** [comment]
Ehhh... not sure if you're done but the logo is different now

image

**Comment by Andrewnt219** [comment]
@vitokhangnguyen Reverted.

File: src/modules/user-auth/components/LoginForm/LoginForm.tsx [hunk 8]
```diff
@@ -76,23 +79,32 @@ const LoginForm = () => {
       <ul aria-label="Sign-in options" tw="space-y-sm">
         <SignInExternalButton
           type="button"
-          icon={<FcGoogle css={signInExternalIconStyle} />}
+          icon={
+            <StyledIconWrapper>
+              <Icon icon={googleFill} tw="text-muted" />
```
Comment (vitokhangnguyen) [inline src/modules/user-auth/components/LoginForm/LoginForm.tsx:84]: Let's use the icons in [this set](https://icon-sets.iconify.design/logos/) for Google, Facebook logo

**Review by vitokhangnguyen — CHANGES_REQUESTED** [review]

**Review by vitokhangnguyen — APPROVED** [review]
Just made a small change to use the Airbnb logo for the "Create a new account" button in the LoginForm. This's gonna be the placeholder for our logo when we have one.

**Pull request merged**